lib/dialects/abstract/query-generator.js

```diff
@@ -1192,9 +1192,10 @@ var QueryGenerator = {
                          primaryKeysLeft[0]
 
             , tableRight = as
+            , keyRight  = association.options ? association.options.otherKey || primaryKeysRight[0] : primaryKeysRight[0]
```

Comment:
You should be able to simply do `association.otherKey` if association is belongs to